lib/ex_aws/cognito_idp.ex

```diff
@@ -0,0 +1,432 @@
+  """
+  @spec admin_user_global_signout(user_pool_id, username) :: op
+  def admin_user_global_signout(user_pool_id, username) do
+    data = %{"UserPoolId" => user_pool_id, "Username" => username}
+
+    request("AdminUserGlobalSignOut", data)
+  end
+
+  # TODO: change_password
+  # TODO: confirm_device
+  # TODO: confirm_forgot_password
+  # TODO: confirm_sign_up
+  # TODO: create_group
+  # TODO: create_identity_provider
+  # TODO: create_resource_server
+  # TODO: create_user_import_job
+  # TODO: create_user_pool
+  # TODO: create_user_pool_client
+  # TODO: create_user_pool_domain
+  # TODO: delete_group
+  # TODO: delete_identity_provider
+  # TODO: delete_resource_server
+  # TODO: delete_user
+  # TODO: delete_user_attributes
+  # TODO: delete_user_pool
+  # TODO: delete_user_pool_client
+  # TODO: delete_user_pool_domain
+  # TODO: describe_identity_provider
+  # TODO: describe_resource_server
+  # TODO: describe_user_import_job
+  # TODO: describe_user_pool
+
+  @spec describe_user_pool(user_pool_id) :: op
+  def describe_user_pool(user_pool_id) do
+    data = %{"UserPoolId" => user_pool_id}
+
+    request("DescribeUserPool", data)
+  end
+
+  # TODO: describe_user_pool_client
+  # TODO: describe_user_pool_domain
+  # TODO: forget_device
+  # TODO: forgot_password
+  # TODO: get_csv_header
+  # TODO: get_device
+  # TODO: get_group
+  # TODO: get_identity_provider_by_identifier
+  # TODO: get_ui_customization
+  # TODO: get_user
+  # TODO: get_user_attribute_verification_code
+  # TODO: global_sign_out
+  # TODO: initiate_auth
+  # TODO: list_devices
+  # TODO: list_groups
+  # TODO: list_identity_providers
+  # TODO: list_user_import_jobs
+  # TODO: list_user_pool_clients
+
+  @type list_users_opts :: [
+          attributes_to_get: [String.t()],
+          filter: String.t(),
+          limit: 0..60
+        ]
+
+  @doc """
+  Lists the users in the Amazon Cognito user pool.
+
+  Can be used with `ExAws.stream!/2` to get all results.
+  """
+  @spec list_users(user_pool_id, list_users_opts) :: op
+  def list_users(user_pool_id, opts \\ []) do
+    data =
+      opts
+      |> Enum.into(%{user_pool_id: user_pool_id})
+      |> camelize_keys()
+
+    stream("ListUsers", "Users", data)
+  end
+
+  # TODO: list_users_in_group
+  # TODO: resend_confirmation_code
+  # TODO: respond_to_auth_challenge
+  # TODO: set_ui_customization
+  # TODO: set_user_settings
+  # TODO: sign_up
+  # TODO: start_user_import_job
+  # TODO: stop_user_import_job
+  # TODO: update_device_status
+  # TODO: update_group
+  # TODO: update_identity_provider
+  # TODO: update_resource_server
+  # TODO: update_user_attributes
+  # TODO: update_user_pool
+  # TODO: update_user_pool_client
+  # TODO: verify_user_attribute
+
+  defp request(action, data) do
+    headers = [
+      {"x-amz-target", "#{@namespace}.#{action}"},
+      {"content-type", "application/x-amz-json-1.1"}
+    ]
+
+    ExAws.Operation.JSON.new(:"cognito-idp", data: data, headers: headers)
```

Comment:
@benwilson512 I tried using `:cognito_idp` as you suggested, but it failed with signature errors:
```elixir
%{
  "__type" => "InvalidSignatureException", 
  "message" => "Credential should be scoped to correct service: 'cognito-idp'. "
}
```

Reply:
I'm thinking we may want to just leave this alone as `:"cognito-idp"`. While slightly less than ergonomic it's rare to actually need to type it, and I'm concerned that it's going to be difficult to try to normalize this at a central place.

Reply:
Yeah, it's just in 3 places, I'm completely fine with leaving it as `:"cognito-idp"`